### PR TITLE
Add the username owning the bucket and impact tests

### PR DESCRIPTION
**Pull request by paraita**

### CODE REVIEW

**Comment by paraita** [comment]
@ow2-proactive/developers 2 reviews please ?

File: src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java [hunk 1]
```diff
@@ -64,8 +67,10 @@
     @RequestMapping(value = "/buckets", method = POST)
     @ResponseStatus(HttpStatus.CREATED)
     public BucketMetadata create(
-            @RequestParam(value = "name", required = true) String bucketName) {
-        return bucketService.createBucket(bucketName);
+            @RequestParam(value = "name", required = true) String bucketName,
+            @ApiParam(value = "The name of the user that owns that Bucket")
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java]: I think it's better "the Bucket" instead of "that Bucket"

**Comment by marcocast** [comment]
:+1:

File: src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java (hunk 1 above)
Comment (paraita) [inline src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java]: true

File: src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java [hunk 2]
```diff
@@ -64,8 +67,10 @@
     @RequestMapping(value = "/buckets", method = POST)
     @ResponseStatus(HttpStatus.CREATED)
     public BucketMetadata create(
-            @RequestParam(value = "name", required = true) String bucketName) {
-        return bucketService.createBucket(bucketName);
+            @RequestParam(value = "name", required = true) String bucketName,
+            @ApiParam(value = "The name of the user that owns the Bucket")
+            @RequestParam(value = "owner") Optional<String> ownerName) {
```
Comment (laurentpellegrino) [inline src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java]: @paraita Why is the owner name optional and not compulsory?

Comment (paraita) [inline src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java]: You're right, and I'm going to run into trouble once I start to work on unicity of bucketName/owner

File: src/integration-test/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketControllerIntegrationTest.java [hunk 3]
```diff
@@ -33,6 +33,7 @@
 
 import java.time.LocalDateTime;
 import java.util.List;
+import java.util.Optional;
```
Comment (laurentpellegrino) [inline src/integration-test/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketControllerIntegrationTest.java]: Is this import still required?

File: src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java [hunk 4]
```diff
@@ -30,11 +30,14 @@
  */
 package org.ow2.proactive.workflow_catalog.rest.controller;
 
+import java.util.Optional;
```
Comment (laurentpellegrino) [inline src/main/java/org/ow2/proactive/workflow_catalog/rest/controller/BucketController.java]: Same as before

File: src/test/java/org/ow2/proactive/workflow_catalog/rest/service/BucketServiceTest.java [hunk 5]
```diff
@@ -60,6 +61,8 @@
     @Mock
     private BucketRepository bucketRepository;
 
+    private final String DefaultBucketName = "BucketServiceTest";
```
Comment (laurentpellegrino) [inline src/test/java/org/ow2/proactive/workflow_catalog/rest/service/BucketServiceTest.java]: static? and snake + uppercase for constants?

File: src/test/java/org/ow2/proactive/workflow_catalog/rest/service/BucketServiceTest.java [hunk 6]
```diff
@@ -44,6 +44,7 @@
 import org.springframework.data.web.PagedResourcesAssembler;
 
 import java.time.LocalDateTime;
+import java.util.Optional;
```
Comment (laurentpellegrino) [inline src/test/java/org/ow2/proactive/workflow_catalog/rest/service/BucketServiceTest.java]: Import required?

**Comment by laurentpellegrino** [comment]
Only a few minor comments, otherwise it looks good to me.

**Pull request merged**